vendor/k8s.io/kubernetes/test/e2e/apimachinery/helpers.go

```diff
@@ -0,0 +1,359 @@
+	err = wait.Poll(500*time.Millisecond, 10*time.Second, func() (done bool, err error) {
+		buf := make([]byte, 128*len(remotePorts))
+		var n int
+		framework.Logf("reading from `kubectl port-forward` command's stdout")
+		if n, err = cmdstdout.Read(buf); err != nil {
+			return false, fmt.Errorf("failed to read from kubectl port-forward stdout: %v", err)
+		}
+		portForwardOutput := string(buf[:n])
+
+		localPorts = make([]int, len(remotePorts))
+		for index, remotePort := range remotePorts {
+			portForwardExpr := fmt.Sprintf("Forwarding from (127.0.0.1|\\[::1\\]):([0-9]+) -> %d", remotePort)
+			portForwardRegexp, err := regexp.Compile(portForwardExpr)
+			if err != nil {
+				return false, fmt.Errorf("failed to compile a regexp for finding a local listening port, err: %v, expression: %s", err, portForwardExpr)
+			}
+			match := portForwardRegexp.FindStringSubmatch(portForwardOutput)
+			if len(match) != 3 {
+				framework.Logf("failed to parse kubectl port-forward output: %q, with %q reg exp, expected to find exactly 3 matches, found: %v", portForwardOutput, portForwardExpr, match)
+				return false, nil
+			}
+
+			localListenPort, err := strconv.Atoi(match[2])
+			if err != nil {
+				return false, fmt.Errorf("error converting %s to an int: %v", match[2], err)
+			}
+			localPorts[index] = localListenPort
+		}
+		if len(localPorts) != len(remotePorts) {
+			framework.Logf("not all ports have been forwarded, found %d forwarded ports but want %d", len(localPorts), len(remotePorts))
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		framework.Failf("Timeout waiting/parsing kubectl port-forward command's output: %v", err)
+	}
+
+	return &portForwardCommand{
+		cmd:        cmd,
+		localPorts: localPorts,
+	}
+}
+
+// apiServersProxyPod creates a pod that once run proxies connections to the given apiServers.
+// It also returns a remote ports list for convenience that will be used by kubectl port-forward
+func apiServersProxyPod(apiServers []string) (*v1.Pod, []int, error) {
+	script := `
+       apk add socat
+       socat TCP-LISTEN:${LOCAL_LISTEN_PORT},fork TCP:${SERVER_IP}:${SERVER_PORT}
+`
+
+	proxyListenPorts := []int{}
+	proxyListenPort := 8443
+	containers := []v1.Container{}
+	for index, apiServer := range apiServers {
+		ipPort := strings.Split(apiServer, ":")
+		if len(ipPort) != 2 {
+			return nil, nil, fmt.Errorf("incorrect apiServer =%s, expected to find an IP address and port in the form of IP:PORT", apiServer)
+		}
+
+		r := strings.NewReplacer(
+			"${LOCAL_LISTEN_PORT}", fmt.Sprintf("%d", proxyListenPort),
+			"${SERVER_IP}", ipPort[0],
+			"${SERVER_PORT}", ipPort[1],
+		)
+		modifiedScript := r.Replace(script)
+
+		containers = append(containers, v1.Container{
+			Name: fmt.Sprintf("server-%d", index),
+			Ports: []v1.ContainerPort{
+				{ContainerPort: int32(proxyListenPort)},
+			},
+			Image:   "alpine", // TODO: build an image with socat and then use  imageutils.GetE2EImage(imageutils.BusyBox),
```

Comment:
do you know how can I get this img upstream?

Reply:
You can just reuse one of the existing e2e images - like agnhost, or debian.

Reply:
`manifest.GetE2EImage(manifest.Agnhost)` etc